src/sync/index.js

```diff
@@ -215,7 +216,14 @@ async function syncTopicCreated(db, startBlock, endBlock, removeHexPrefix) {
         const insertTopicDB = new Promise(async (resolve) => {
           try {
             const topic = new Topic(blockNum, txid, rawLog).translate();
-            DBHelper.insertOrUpdateTopic(db.Topics, topic);
+
+            // Update existing mutated Topic or insert new
+            if (DBHelper.getCount(db.Topics, { txid }) > 0) {
+              DBHelper.updateTopicByQuery(db.Topics, topic, { txid });
```

Comment:
@dwalintukan is this update for safety? we should skip this insertion

Reply:
This is necessary. We will get duplicated Topics if we just do insertion. `db.insert` does not handle updating. Or does it? 

If I created this topic, the Topic will already be in the DB, so when we parse the searchlog entry we need to update that specific one.

Reply:
@dwalintukan i mean line 222 can be skipped while keeping the `if` condition, because txid has been updated,. any other fields to update?

Reply:
We also need update: `status`, `address`, `blockNum`